apollo-router/tests/integration_tests.rs

```diff
@@ -66,6 +66,9 @@ async fn basic_composition() {
 }
 
 #[test_span(tokio::test)]
+#[target(reqwest=tracing::Level::INFO)]
+#[target(reqwest_tracing=tracing::Level::INFO)]
+#[target(hyper::client=tracing::Level::INFO)]
```

Comment:
Don't we actually want:
```
#[level(tracing::Level::ERROR)]
#[target(apollo_router=tracing::Level::DEBUG)]
#[target(apollo_router_core=tracing::Level::DEBUG)]
```
to reduce the prospect of interference?

The above should mean we only get messages at level ERROR from things which aren't apollo_router or apollo_router_core. I believe that's the goal, but I could be on the wrong track...

Reply:
Hmm that's a very interesting comment!

the default level for all logs in test_span is debug, unless specified otherwise, which means this tests kind of unfolds to:

```
#[levelreqwest=tracing::Level::DEBUG)]
#[target(reqwest=tracing::Level::INFO)]
#[target(reqwest_tracing=tracing::Level::INFO)]
#[target(hyper::client=tracing::Level::INFO)]
```

which effectively means we're tracking spans and logs at the `DEBUG` level unless it's from `reqwest`, `reqwest_tracing`, and `hyper`.

What teases my mind is that your comment may call for a `workspace level` and a `dependencies level`, which I'll happily open an issue for in test_span, if it sounds good to you?

Reply:
The generally accepted default for log/tracing levels is INFO. That means it's a bit "surprising" (as in principle of least surprise) that test_span's default level is DEBUG.

That aside, I think that if we were going to vary from INFO as a default, then we should probably vary to ERROR, to make logging an (almost) "opt-in" decision and thus attempt to minimize random variability when new dependencies are added.

I'd probably prefer: default is INFO (so no surprises) and also recommend that people configure a default of ERROR (as I did) and opt-in for the stuff they want to capture rather than specifying what they want to exclude (which is more fragile and likely to break as crates are added/removed).

My second preference would be to set the default to ERROR and again require an opt-in for stuff you are interested in.

Reply:
I feel like defaulting to INFO would make a lot of sense indeed, I'm not quite sure ERROR would be a sensible default for a span / logs snapshots crate. I'm opening an issue!

Reply:
But wouldn't you just add an entry for `apollo_rs`? e.g.:
```
#[target(apollo_rs=tracing::Level::DEBUG)]
```

Reply:
that works as well! https://github.com/apollographql/router/blob/d088afdc914bf264efa447be569e3cac5b5948d8/apollo-router/tests/integration_tests.rs#L69